### PR TITLE
Home Page Responsiveness + Sample Census Formatting

### PR DESCRIPTION
**Pull request by ianHunterClipd**
Hey all, I am working on issues #78 and #73 and was hoping to get some feedback on the code before I proceed. 

Edit - this also takes care of #94 

Re: responsiveness - I am controlling what is presented by doing direct comparisons with window.innerWidth (example in DetailViewContainer.js, lines 356 - 396). Is this a valid approach or would you recommend some other strategy? 

Re: formatting the census question subtexts. I am unclear on when / where I should be using the FormattedMarkdownMessage. I played with the component and was able to get it to render some markdown but I feel like I am missing something. Any comments on SampleCensus.js and the definition of the questions array (starts @ line 10) would be appreciated.

### CODE REVIEW

File: client/components/DetailViewContainer.js [hunk 1]
```diff
@@ -341,7 +353,47 @@ export default class DetailViewContainer extends Component {
               </Link>
             }
           />
+          { screen_width < 800 && (
```
Comment (ianHunterClipd) [inline client/components/DetailViewContainer.js:356]: This is an example of my comparisons with window.innerWidth (saved locally as screen_width) for responsiveness.

File: client/components/FormattedMarkdownMessage.js [hunk 2]
```diff
@@ -4,6 +4,7 @@ import PropTypes from 'prop-types';
 import Markdown from 'react-remarkable';
 
 const FormattedMarkdownMessage = ({ intl, id, description, defaultMessage, values }) => {
+  console.log(values)
```
Comment (JMStudiosJoe) [inline client/components/FormattedMarkdownMessage.js]: remove the logs please unless needed there for debugging?

**Review by JMStudiosJoe — CHANGES_REQUESTED** [review]
looks good so far only need to remove logs.

**Comment by turnerhayes** [comment]
The ideal approach to responsiveness is using CSS with [media queries](https://developer.mozilla.org/en-US/docs/Web/CSS/Media_Queries/Using_media_queries); these let you apply certain styles only if certain conditions are met (e.g. the device width is > 800px). This is better from both a separation of concerns perspective, and, I think, from a performance perspective (I am under the impression that applying new styles is generally less expensive than manipulating DOM).

As for FormattedMarkdownMessage, unfortunately there are some caveats to using it, due to the markdown parsing library we're using (I describe them in #98). Use it in cases where the content you're rendering is already meant to be displayed as a paragraph (or series of paragraphs); e.g. good for FAQ answers, not good for headers.

File: client/components/DetailViewContainer.js [hunk 3]
```diff
@@ -102,28 +103,93 @@ CarouselItem.propTypes = {
...
+  } else {
+    size = 'large'
+  }
+  let top_reasons_class = "c_home__top-reasons"
+
+  return (
+    <ol
+      className={classnames(
+        top_reasons_class,
+        `${top_reasons_class}__${size}`,
```
Comment (turnerhayes) [inline client/components/DetailViewContainer.js:143]: Good application of BEM! One minor thing, the "official" [BEM specs](http://getbem.com/naming/) use `--` as a separator for modifiers like ${size}. You could change this to `${top_reasons_class}--${size}`.

File: client/components/SampleCensus.js [hunk 4]
```diff
@@ -7,17 +7,195 @@ import { FormattedMarkdownMessage } from './FormattedMarkdownMessage';
...
+      </FormattedMessage>
+    ),
+    secondary_information: (
+      <ul
+        style={{
+          display: 'flex',
+          flexDirection: 'column',
+          padding: '0',
+        }}>
+        { /*<FormattedMarkdownMessage
```
Comment (turnerhayes) [inline client/components/SampleCensus.js:65]: Remove commented out code

**Review by turnerhayes — APPROVED** [review]
I'm okay with this going through as is, but a couple of things:

* It would be better to use media queries for the responsiveness; if you change the DOM or inline styles, React will rerender the subtree, which we want to minimize. If you use CSS, the DOM doesn't have to change when screen size changes.
* I would prefer to have the content stuff (sample census) be in their on PR, if that wouldn't be too difficult.

**Comment by ianHunterClipd** [comment]
@turnerhayes 
Re media queries - got it. 

Re sample census - I'll remove the existing two questions from the sample census from this PR and move them into a new one.

**Review by JMStudiosJoe — APPROVED** [review]

**Pull request merged**